src/main/java/org/gridsuite/loadflow/server/repositories/LoadFlowResultRepository.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.loadflow.server.repositories;
+
+import com.powsybl.loadflow.LoadFlowResult;
+import org.gridsuite.loadflow.server.entities.ComponentResultEntity;
+import org.gridsuite.loadflow.server.entities.GlobalStatusEntity;
+import org.gridsuite.loadflow.server.entities.LoadFlowResultEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * @author Anis Touri <anis.touri at rte-france.com
+ */
+@Repository
+public class LoadFlowResultRepository {
+
+    private GlobalStatusRepository globalStatusRepository;
+
+    private ResultRepository resultRepository;
+
+    public LoadFlowResultRepository(GlobalStatusRepository globalStatusRepository,
+                                    ResultRepository resultRepository) {
+        this.globalStatusRepository = globalStatusRepository;
+        this.resultRepository = resultRepository;
+    }
+
+    private static LoadFlowResultEntity toResultEntity(UUID resultUuid, LoadFlowResult result) {
+        Set<ComponentResultEntity> componentResults = result.getComponentResults().stream().map(LoadFlowResultRepository::toComponentResultEntity).collect(Collectors.toSet());
+        return new LoadFlowResultEntity(resultUuid, ZonedDateTime.now(ZoneOffset.UTC), componentResults);
+    }
+
+    private static ComponentResultEntity toComponentResultEntity(LoadFlowResult.ComponentResult componentResult) {
+        int connectedComponentNum = componentResult.getConnectedComponentNum();
```

Comment:
You can avoid declaring all those temporary variables I think

Reply:
done